src/deadline/job_attachments/download.py

```diff
@@ -1213,3 +1233,138 @@ def download_job_output(
         progress_tracker.total_time = time.perf_counter() - start_time
 
         return progress_tracker.get_download_summary_statistics(downloaded_files_paths_by_root)
+
+
+def _download_job_manifests_using_s3_keys(
+    session: boto3.Session,
+    manifest_keys: List[str],
+    job_attachment_settings: JobAttachmentS3Settings,
+    download_directory: str,
+) -> None:
+    """
+    Downloads job manifests from S3 to a local directory using provided S3 keys.
+
+    Note: Current implementation downloads sequentially. Future optimization
+    opportunity exists for parallel downloads.
+
+    Args:
+        session: boto3 Session
+        manifest_keys: List of S3 object keys for manifest files
+        job_attachment_settings: S3 job attachments settings
+        download_directory: Local directory path where manifests will be saved
+
+    Raises:
+        JobAttachmentsError: If manifest keys are malformed or if there are IO errors
+        JobAttachmentS3BotoCoreError: If there are AWS S3 errors (excluding missing files)
+    """
+    s3 = get_s3_client(session=session)
+
+    for manifest_key in manifest_keys:
+        split_key = manifest_key.split("/")
+
+        if len(split_key) < 2:
+            raise JobAttachmentsError(
+                f"Invalid manifest key structure: {manifest_key}. Expected at least 2 path segments."
+            )
+
+        file_name = f"{split_key[-2]}_{split_key[-1]}"
+        local_file_path = os.path.join(download_directory, file_name)
+
+        try:
+            s3.download_file(job_attachment_settings.s3BucketName, manifest_key, local_file_path)
+        except (ClientError, IOError) as err:
+            raise JobAttachmentsError(f"Failed to download manifest {manifest_key}: {err}") from err
+
+
+def _get_all_manifest_s3_keys_for_job(
+    session: boto3.Session,
+    job_attachment_settings: JobAttachmentS3Settings,
+    farm_id: str,
+    queue_id: str,
+    job_id: str,
+) -> List[str]:
+    """
+    Retrieves all manifest (both input and output) S3 keys for a specific job.
+
+    Args:
+        session: boto3 Session for AWS credentials
+        job_attachment_settings: S3 job attachments settings
+        farm_id: Deadline farm identifier
+        queue_id: Deadline queue identifier
+        job_id: Deadline job identifier
+
+    Returns:
+        List[str]: Combined list of input and output manifest S3 keys
+
+    Raises:
+        JobAttachmentsError: If there's any error retrieving the manifests
+    """
+    root_prefix: str = job_attachment_settings.rootPrefix.rstrip("/")
+    output_manifest_prefix: str = f"{root_prefix}/Manifests/{farm_id}/{queue_id}/{job_id}/"
+
+    try:
+        input_manifest_keys: List[str] = _get_input_manifest_keys(
```

Comment:
nit - what about having `from_job` to following `_get_tasks_manifests_keys_from_s3` format and provide more context?

```suggestion
        input_manifest_keys: List[str] = _get_input_manifest_keys_from_job(
```

Reply:
Good point. I changed the name to `_get_input_manifest_keys_for_job`. I think "for" makes more sense than "from" here.